infrastructure/src/main/kotlin/DetektPlugin.kt

```diff
@@ -52,6 +54,26 @@ private fun Project.configureDetektTasks(extension: RedmadrobotExtension) {
     detektTask(extension, "detektAll") {
         description = "Runs over whole code base without the starting overhead for each module."
     }
+
+    if (extension.detekt.checkDiffOnly && extension.detekt.targetDiffBranch.isNotEmpty()) {
+        val changedFilesFilter = FilterParams(
+            changeTypes = listOf(ChangeType.ADD, ChangeType.MODIFY),
+            fileExtensions = listOf(".kt")
+        )
+
+        val findChangedFiles = tasks.register<FindChangedFilesTask>(
+            name = "findChangedFiles",
+            extension.detekt.targetDiffBranch,
+            changedFilesFilter
+        ).apply {
+            configure { projectDirProperty.set(layout.projectDirectory) }
+        }
```

Comment:
Without injection it will look better:
```kotlin
val findChangedFiles = tasks.register<FindChangedFilesTask>("findChangedFiles") {
    projectDirProperty.set(layout.projectDirectory)
    baseBranch = extension.detekt.targetDiffBranch
    filesFilter = changedFilesFilter
}
```

Btw, what about renaming `targetBranch` to `baseBranch` as it [called in GitHub](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-base-branch-of-a-pull-request)?

Reply:
Renamed.